bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,174 @@
+import org.openhab.binding.revogi.internal.api.StatusService;
+import org.openhab.binding.revogi.internal.api.SwitchService;
+import org.openhab.binding.revogi.internal.udp.DatagramSocketWrapper;
+import org.openhab.binding.revogi.internal.udp.UdpSenderService;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RevogiSmartStripControlHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class RevogiSmartStripControlHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RevogiSmartStripControlHandler.class);
+    private final StatusService statusService;
+    private final SwitchService switchService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable RevogiSmartStripControlConfiguration config;
+
+    public RevogiSmartStripControlHandler(Thing thing) {
+        super(thing);
+        UdpSenderService udpSenderService = new UdpSenderService(new DatagramSocketWrapper(), scheduler);
+        this.statusService = new StatusService(udpSenderService);
+        this.switchService = new SwitchService(udpSenderService);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RevogiSmartStripControlBindingConstants.PLUG_1_SWITCH:
+                switchPlug(command, 1);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_2_SWITCH:
+                switchPlug(command, 2);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_3_SWITCH:
+                switchPlug(command, 3);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case RevogiSmartStripControlBindingConstants.ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+            default:
+                logger.debug("Something went wrong, we've got a message for {}", channelUID.getId());
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        @Nullable
```

Comment:
The compiler is more intelligent on local variables. Therefore, the Nullable could be removed.

Reply:
done